src/core/connection.c

```diff
@@ -5626,6 +5626,30 @@ QuicConnParamSet(
         Status = QUIC_STATUS_SUCCESS;
         break;
 
+    case QUIC_PARAM_CONN_SSLKEYLOG_BUFFER:
+#ifdef QUIC_WIRESHARK_SUPPORT
+
+        if (BufferLength == 0 || Buffer == NULL) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        if (Connection->State.Started) {
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
+
+        Connection->SslKeyLogBuffer = (uint8_t*)Buffer;
```

Comment:
Instead of a generic buffer, I'd rather standardize this around a type specific struct that contains arrays for each key type. Then, the connection has the (optional) pointer to the struct. If it's set, then it will write the struct arrays, per key.